community/cypher/util-3.4/src/main/scala/org/neo4j/cypher/internal/util/v3_4/Rewritable.scala

```diff
@@ -34,7 +34,11 @@ object Rewriter {
 
 object RewriterWithArgs {
   def lift(f: PartialFunction[(AnyRef, Seq[AnyRef]), AnyRef]): RewriterWithArgs =
-    f.orElse(PartialFunction({ case (a: AnyRef, _) => a }))
+    f.orElse(PartialFunction({
+      // We need to dup anything not matched by f given the children
+      case (p: Product, children) => new DuplicatableProduct(p).dup(children).asInstanceOf[AnyRef]
+      case (a: AnyRef, children) => new DuplicatableAny(a).dup(children)
```

Comment:
Arguably we no longer need implicit classes here, but could just call functions instead with the instance as argument?